### Задача 1

Контрольная работа состоит из пяти вопросов. На каждый вопрос приведено четыре варианта ответа, один из которых правильный. Составьте закон распределения числа правильных ответов при простом угадывании. Найдите M(X), D(X).

### Решение

Пусть X - случайная величина, равная числу правильных ответов. Она
распределена по биномиальному закону с параметрами n = 5 (число вопросов), 
p = 0.25 (вероятность угадать ответ), поэтому вероятности вычисляются по формуле Бернулли 
$$P(X = k) = C_n^k p^k (1 - p)^{n - k}$$


In [2]:
import numpy as np

In [3]:
def combinations(n: int, k: int) -> int:
    """Число сочетаний.
    """
    return np.math.factorial(n) // (np.math.factorial(k) * np.math.factorial(n - k))

In [ ]:
Вероятность получения 5 верных ответов 

In [7]:
combinations(5,5)*np.power((1/4),5)*np.power((3/4),0)=

0.0009765625

In [ ]:
Вероятность получения 4 верных ответов = 15/4 ** 5

In [11]:
combinations(5,4)*np.power((1/4),4)*np.power((3/4),1)

0.0146484375

Для 3 верных ответов:
    

In [13]:
combinations(5,3)*np.power((1/4),3)*np.power((3/4),2)

0.087890625

Для 2 верных ответов:

In [14]:
combinations(5,2)*np.power((1/4),2)*np.power((3/4),3)

0.263671875

Для 1 верного ответа:
    

In [15]:
combinations(5,1)*np.poНиwer((1/4),1)*np.power((3/4),4)

0.3955078125

Вероятность ни одного верного ответа

In [18]:
combinations(5,0)*np.power((1/4),0)*np.power((3/4),5)

0.2373046875

#### Закон распределения:

<table border="3">
    <tr>
        <th>$x$</th>
        <td>0</td>
        <td>1</td>
        <td>2</td>
        <td>3</td>
        <td>4</td>
        <td>5</td>
    </tr>
    <tr>
        <th>$P(X=x)$</th>
        <td>0.2373046875</td>
        <td>0.3955078125</td>
        <td>0.263671875</td>
        <td>0.087890625</td>
        <td>0.0146484375</td>
        <td>0.0009765625</td>
    </tr>
</table>


Математическое ожидание по ф-ле Бернулли М = n * p


###### M = 6*1/4 = 1.5

###### Дисперсия  D = n * p * (1-p)     6 * 1 .4 * (1 -1 / 4) = 1.125

### Задача 2

Вероятность попадания в цель при одном выстреле равна 0.01. Какова вероятность того, что число попаданий при 200 выстрелах будет не менее 5 и не более 10?

Пусть  X – число попаданий в цель. Так как вероятность p=0.01 очень мала, а число выстрелов большое, то  будем использовать формулу Пуассона $$P(X = k) = \dfrac{a^k e ^{-a}}{k!}.$$  и сложения вероятностей

a = n * p = 200* 0.01 = 2, а k - число попаданий в цель

$$P(X) = e^{-2}*(\dfrac{2^5}{5!} + \dfrac{2^6}{6!}+ \dfrac{2^7}{7!}+ \dfrac{2^8}{8!}+ \dfrac{2^9}{9!}+ \dfrac{2^{10}}{10!}).$$

In [4]:
def pois(n: int) :
    return np.power(2,n)/np.math.factorial(n)

Вероятность равна:

In [6]:
s=0
for n in range(5,11):
    s+=pois(n)
print(s*np.exp(-2))

0.05264470911934268


### Задача 3

В условиях предыдущей задачи сколько выстрелов нужно сделать, чтобы быть уверенным с вероятностью 0.9, что хотя бы 10 раз будет совершено попадание?

Переформулировка: сколько выстрелов нужно сделать, чтобы с вероятностью 0.1 попадание было не более 10 раз ? (Вероятность попадания в цель = 0.01) То есть для одного случая надо рассматривать сумму вероятностей из попаданий 1 раз, 2 раза и т.д. - до 10раз. Вероятность рассчитываем по Пуассону
    

In [22]:
def prob(n,p,k):
    return ((n*p)**k)*np.exp(-n*p)/np.math.factorial(k)

In [29]:
n = 100 #начальное кол-во выстрелов (должно быть большим)
p = 5 
while(p > 0.1):
    n += 1
    p = 0
    for k in range(10):
      # print (p)
        p += prob(n, 0.01, k)
      
     

In [30]:
print('Количество выстрелов:', n)

Количество выстрелов: 1421
